src/QsCompiler/Compiler/RewriteSteps/SyntaxTreeTrimming.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.SyntaxTreeTrimming;
+
+namespace Microsoft.Quantum.QsCompiler.BuiltInRewriteSteps
+{
+    /// <summary>
+    /// Removes unused callables from the syntax tree.
+    /// </summary>
+    internal class SyntaxTreeTrimming : IRewriteStep
+    {
+        private readonly bool keepAllIntrinsics;
+
+        public string Name => "Syntax Tree Trimming";
+
+        public int Priority => RewriteStepPriorities.SyntaxTreeTrimming;
+
+        public IDictionary<string, string?> AssemblyConstants { get; } = new Dictionary<string, string?>();
+
+        public IEnumerable<IRewriteStep.Diagnostic> GeneratedDiagnostics => Enumerable.Empty<IRewriteStep.Diagnostic>();
+
+        public bool ImplementsPreconditionVerification => true;
+
+        public bool ImplementsTransformation => true;
+
+        public bool ImplementsPostconditionVerification => false;
+
```

Comment:
1. (My subjective opinion, feel free to ignore)
I prefer more dense code, code that occupies less space on the screen. I would not use empty lines here.
But if that is the requirement of the coding style then I'm totally fine with that.
2. (Minor) I suspect some of these are reasonable to be `const` (speaking in C++), or `readonly`? (speaking in C#).

Reply:
OK. (I don't know C#, looks like these are not member _vars_ but are member _functions_... OK)

Reply:
As a side note, C# does have the `const` keyword as well as `readonly`. In particular, `const` denotes a compile-time constant, while a `readonly` field can only be set during a constructor; after that constructor returns, the field can no longer be modified.